packages/eslint-scope/lib/referencer.js

```diff
@@ -649,6 +649,45 @@ class Referencer extends esrecurse.Visitor {
 
         // do nothing.
     }
+
+    JSXIdentifier(node) {
+        if (this.scopeManager.__isJSXEnabled()) {
+            this.currentScope().__referencing(node);
+        }
+    }
+
+    JSXMemberExpression(node) {
+        this.visit(node.object);
+    }
+
+    JSXElement(node) {
+        if (this.scopeManager.__isJSXEnabled()) {
+            this.visit(node.openingElement);
+            node.children.forEach(this.visit, this);
+        } else {
+            this.visitChildren(node);
+        }
+    }
+
+    JSXOpeningElement(node) {
+        const name = node.name.name;
+
+        if (this.scopeManager.__isJSXEnabled() && name[0].toUpperCase() === name[0]) {
```

Comment:
This would crash if `node.name` is not a `JSXIdentifier`. For example: `<obj.prop/>`.